### PR TITLE
exercises

### PR DESCRIPTION
**Pull request by anbaopham**

### CODE REVIEW

File: students/anbaopham/temp/lesson_2_task_1.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            print("+" + "-"*a + "+" + "-"*a + "+", end= ' ')
+        else:
+            print("|" + " "*a + "|"+ " "*a + "|")
+
+
+def print_grid2(c,x):
+    a = (x-1)//2
+    line_1 = ("+" + "-"*x)*(c-1) + ("+" + "-"*x+"+")
+    line_2 = ("|"+ " "*x)*(c-1) + ("|") + " "*x + "|")
+    for i in range(3*x+2):
```
Comment (charlesrobison) [inline students/anbaopham/temp/lesson_2_task_1.py:18]: Is this what you intended?  Doesn't seem to be correct

File: students/anbaopham/temp/lesson_2_task_1.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    a = (x-1)//2
+    for i in range(x+2):
+        if i ==0:
+            print("+" + "-"*a + "+" + "-"*a + "+")
+        elif i == a+1:
+            print("+" + "-"*a + "+" + "-"*a + "+")
+        elif i == x+1:
+            print("+" + "-"*a + "+" + "-"*a + "+", end= ' ')
+        else:
+            print("|" + " "*a + "|"+ " "*a + "|")
```
Comment (charlesrobison) [inline students/anbaopham/temp/lesson_2_task_1.py:11]: Spacing is inconsistent in this file

File: students/anbaopham/temp/lesson_2_task_1.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            print(line_1)
+
+        elif i== x+1+b:
+            print(line_1)
+            b = i
+
+        else:
+            print(line_2)
+
+def FizzBuzz(x=100):
```
Comment (charlesrobison) [inline students/anbaopham/temp/lesson_2_task_1.py:44]: This should be snake_case

File: students/anbaopham/temp/lesson_2_task_1.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    for i in range(1,x,1):
+        if i%3 == 0:
+            print("Fizz")
+        elif i%5 ==0:
+            print("Buzz")
+        else:
+            print(i)
```
Comment (charlesrobison) [inline students/anbaopham/temp/lesson_2_task_1.py:51]: Only goes to 99

File: students/anbaopham/lesson06/test_Assignment_4.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+          ">>> "))
+
+def user_input_full_name():
+    while True:
+        full_name = input("Pleass enter full name >> ")
+        full_name = full_name.strip()
+
+        if len(full_name) == 0:
+            print("please try again")
+        elif (' ' in full_name) == False:
```
Comment (charlesrobison) [inline students/anbaopham/lesson06/test_Assignment_4.py:35]: should be is False:

File: students/anbaopham/lesson06/test_Assignment_4.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+def add_donation_amount(full_name, donation_amount):
+    for i in donor_db:
+
+        if full_name == i[0]:
+            i[1].append(donation_amount)
+
+def test_add_donation_amount():
+    full_name ="a b"
+    donation_amount=100
```
Comment (charlesrobison) [inline students/anbaopham/lesson06/test_Assignment_4.py:73]: donation_amount is assigned but isn't used.  Also, needs spaces

File: students/anbaopham/lesson06/test_Assignment_4.py [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+    l1 =sorted(l1, key = sort_total_given)
+    l1.insert(0,l)
+
+    new_list = []
+
+    [new_list.append(len(j)) for i in l1 for j in i]
+    l = max(new_list)
+    space = " "*10
+    width = l
+    print(l1)
```
Comment (charlesrobison) [inline students/anbaopham/lesson06/test_Assignment_4.py:225]: Sorting isn't working - should be sorted by total given in descending order

File: students/anbaopham/lesson06/test_Assignment_4.py [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+        #except:
+        #    print("Not a valide option!")
+
+
+
+
+if __name__ == "__main__":
+    # don't forget this block to guard against your code running automatically if this module is imported
+
+    main()
```
Comment (charlesrobison) [inline students/anbaopham/lesson06/test_Assignment_4.py:255]: There should be a separate file with unit tests for this assignment.  Do you have it?

File: students/anbaopham/lesson08/circle.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+    def __str__(self):
+        return "Circle with radius: {}".format(self.radius)
+
+    def __repr__(self):
+        return "Circle({})".format(self.radius)
+
+    def __add__(self, other):
+        self.radius += other.radius
+        return Circle(self.radius)
```
Comment (charlesrobison) [inline students/anbaopham/lesson08/circle.py:36]: Need a blank line here.  Always a blank line between functions, everywhere in the code

**Review by charlesrobison — APPROVED** [review]
Please review comments and let me know if you have questions.

@natasha-aleksandrova - only lesson 9 left to review, otherwise ready to merge

File: students/anbaopham/lesson09/donor_models.py [hunk 10]
```diff
@@ -0,0 +1,86 @@
+import sys
+
+class DonorCollection(object):
```
Comment (natasha-aleksandrova) [inline students/anbaopham/lesson09/donor_models.py:3]: the relationship in DonorCollection and Donor classes isn't correct.

My suggestion is to review the instructions and tutorial for more help.

I also have this stub to demonstrate it more hands on:
https://repl.it/@naaleksandrova/Lesson9Example

**Pull request merged**